packet/src/pacer.rs

```diff
@@ -0,0 +1,967 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::time::{Duration, Instant};
+
+use rtp::{Bitrate, DataSize};
+
+const MAX_BITRATE: Bitrate = Bitrate::gbps(10);
+const MAX_MEDIA_DEBT_IN_TIME: Duration = Duration::from_millis(500);
+
+/// A packet Pacer.
+///
+/// The pacer is responsible for ensuring correct pacing of packets onto the network at a given
+/// bitrate.
+pub trait Pacer {
+    /// Set the target bitrate to send at. This can be exceeded if required.
+    fn set_target_bitrate(&mut self, target_bitrate: Bitrate);
```

Comment:
Initially I did not expect this. I thought it was not needed because sr0m knows (measures) the bitrate of the streams flowing, and it's more realistic compared to a supposed target bitrate - especially in the case of streams like screen shares which can diverge a lot from the target.
Then I also thought this could be misleading as someone could think this is the target bitrate you want to test for (the one you want to reach) and not the current one.
All in all, I don't think these are blockers at our stage.

Reply:
libwebrtc calls it the "pacing rate" or "data rate"and definitely has a setter for it.  It also separately has a "padding rate" or "pad rate".   These rates are based on the output of the congestion controller, but they are *not* just the target send rate from the congestion controller.  For example, by default WebRTC applies 1.1 "pacing factor". 

So, I would call this set_pacing_bitrate or set_pacing_bitrates with a TODO to have a separate padding bitrate.